### PR TITLE
feat: api plugin from scratch templates

### PR DESCRIPTION
**Pull request by huimiu**
Feature: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27000310

- Include JS, TS and CSharp templates.
- Build the template using the repair API (no auth).
- The appPackage/ai-plugin.json is implemented based on [this schema definition](https://spec-hub.azurewebsites.net/specifications/PluginManifest-2.0.html).

### CODE REVIEW

File: templates/ts/api-plugin-from-scratch/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Overview of API Plugin from New API Template
+
```
Comment (MuyangAmigo) [inline templates/ts/api-plugin-from-scratch/README.md:2]: Title: Overview of the Copilot Plugin template

File: templates/ts/api-plugin-from-scratch/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Build an API Plugin from a new API with Azure Functions
+
```
Comment (MuyangAmigo) [inline templates/ts/api-plugin-from-scratch/README.md:4]: Build a Copilot Plugin from a new API....

File: templates/ts/api-plugin-from-scratch/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+This app template allows Teams to interact directly with third-party data, apps, and services, enhancing its capabilities and broadening its range of capabilities. It allows Teams to:
+
+- Retrieve real-time information, for example, latest news coverage on a product launch.
```
Comment (MuyangAmigo) [inline templates/ts/api-plugin-from-scratch/README.md:7]: This app template allows Microsoft Copilot for Microsoft 365 to interact....

File: templates/ts/api-plugin-from-scratch/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+## Get started with the template
+
+> **Prerequisites**
+>
+> To run this app template in your local dev machine, you will need:
+>
+> - [Node.js](https://nodejs.org/), supported versions: 16, 18
+> - A [Microsoft 365 account for development](https://docs.microsoft.com/microsoftteams/platform/toolkit/accounts)
+> - [Teams Toolkit Visual Studio Code Extension](https://aka.ms/teams-toolkit) version 5.0.0 and higher or [Teams Toolkit CLI](https://aka.ms/teamsfx-cli)
+
```
Comment (MuyangAmigo) [inline templates/ts/api-plugin-from-scratch/README.md:19]: One more thing in Prerequisites is Copilot License:

- [Copilot for Microsoft 365 license](https://learn.microsoft.com/microsoft-365-copilot/extensibility/prerequisites#prerequisites)

File: templates/ts/api-plugin-from-scratch/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+1. First, select the Teams Toolkit icon on the left in the VS Code toolbar.
+2. In the Account section, sign in with your [Microsoft 365 account](https://docs.microsoft.com/microsoftteams/platform/toolkit/accounts) if you haven't already.
+3. Select `Debug in Teams (Edge)` or `Debug in Teams (Chrome)` from the launch configuration dropdown.
+4. Send a message to Copilot to find a repair record.
+
+## What's included in the template
```
Comment (MuyangAmigo) [inline templates/ts/api-plugin-from-scratch/README.md:25]: Let's add a screenshot when possible

File: templates/ts/api-plugin-from-scratch/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+The following files can be customized and demonstrate an example implementation to get you started.
+
+| File                                         | Contents                                                            |
+| -------------------------------------------- | ------------------------------------------------------------------- |
+| `src/functions/repair.ts`                    | The main file of a function in Azure Functions.                     |
+| `src/repairsData.json`                       | The data source for the repair API.                                 |
+| `appPackage/apiSpecificationFile/repair.yml` | A file that describes the structure and behavior of the repair API. |
+| `appPackage/ai-plugin.json`                  | The manifest file for the API plugin.                               |
+
```
Comment (MuyangAmigo) [inline templates/ts/api-plugin-from-scratch/README.md:43]: We should list bot Teams and Type B json manifest here as these are equally important.

* manifest.json: Teams application manifest that defines metadata for your plugin inside Microsoft Teams.
* ai-plugin.json: The manifest file for your Copilot Plugin that contains information for your API and used by LLM.

**Review by MuyangAmigo — CHANGES_REQUESTED** [review]

File: templates/ts/api-plugin-from-scratch/README.md (hunk 5 above)
Comment (huimiu) [reply templates/ts/api-plugin-from-scratch/README.md:26]: Sure, I will add a screenshot once the E2E test is passed.

File: templates/ts/api-plugin-from-scratch/README.md (hunk 6 above)
Comment (huimiu) [reply templates/ts/api-plugin-from-scratch/README.md:45]: All the changes you suggested have been made, thanks.

**Review by MuyangAmigo — APPROVED** [review]

File: templates/js/api-plugin-from-scratch/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+- Retrieve real-time information, for example, latest news coverage on a product launch.
+- Retrieve knowledge-based information, for example, my team’s design files in Figma.
+
+## Get started with the template
+
+> **Prerequisites**
+>
+> To run this app template in your local dev machine, you will need:
+>
+> - [Node.js](https://nodejs.org/), supported versions: 16, 18
```
Comment (hund030) [inline templates/js/api-plugin-from-scratch/README.md:16]: Is the version correct?

**Review by hund030 — DISMISSED** [review]

File: templates/js/api-plugin-from-scratch/README.md (hunk 7 above)
Comment (huimiu) [reply templates/js/api-plugin-from-scratch/README.md]: Updated, thanks!

**Review by hund030 — DISMISSED** [review]

**Review by eriolchan — APPROVED** [review]

**Pull request merged**